### ISSUE TITLE
App plugin: cannot display react page content because of angular issue

### ISSUE THREAD

**Issue by toussa** [grafana/grafana#22220]
**What happened**:
When trying to develop an app plugin with react pages:

React pages of an app plugin cannot be displayed and the following angular error is displayed in console:
```
in ChangeTracker.ts:6
Plugin component error Error: [$injector:unpr] Unknown provider: propsProvider <- props
http://errors.angularjs.org/1.6.9/$injector/unpr?p0=propsProvider%20%3C-%20props
    at angular.js:116
    at angular.js:4887
    at Object.s [as get] (angular.js:5047)
    at angular.js:4892
    at s (angular.js:5047)
    at u (angular.js:5072)
    at Object.invoke (angular.js:5098)
    at M.instance (angular.js:11076)
    at it (angular.js:9926)
    at angular.js:9248
```

**What you expected to happen**:
React pages should be displayed correctly.

**How to reproduce it (as minimally and precisely as possible)**:

Using the grafana/simple-app-plugin and trying to set the root page to the file ExampleRootPage.tsx
To do so, add the page entry to includes array:

```
"includes": [
    {
      "type": "page",
      "name": "React Page",
      "component": "ExampleRootPage",
      "role": "Viewer",
      "addToNav": true,
      "defaultNav": true
    },
...
```
and in module.ts:

```
export {
  ExampleConfigCtrl as ConfigCtrl,
  // AngularExamplePageCtrl,
  ExampleRootPage // Must match `pages.component` in plugin.json
};

export const plugin = new AppPlugin<ExampleAppSettings>()
  .setRootPage(ExampleRootPage)
```

**Anything else we need to know?**:
See other links that may be related to this issue:
https://github.com/grafana/simple-app-plugin/issues/5
https://community.grafana.com/t/react-pages-in-app-plugin/

**Environment**:
- Grafana version: 6.6.1
- Data source type & version: none
- OS Grafana is installed on: alpine & ubuntu (docker)
- User OS & Browser: chrome & debian
- Grafana plugins: simple-app-plugin
- Others:

**Comment by gopalakrishnan-subramani** [comment]
This issue has been reported by many times, seems like there is a gap in understanding core issues.

We have two types of pages when it comes to app plugins modules, One is Config Page, which appear inside the tabs, which works fine for AngularJS, React page types.

When it comes to normal pages, which doesn't come under the tab config page, instead shown independent. We can launch the same from left side app menu options. Only angular pages works, not the react pages.

**Comment by toussa** [comment]
Thanks.
Do you know any workaround ?

Other issues are either closed (like #21390), talking about side navigation bar or mixing other issue (like angular pages inconsistency) 
I think this issue nails down only one problem, which I find quite important.

**Comment by papagian** [comment]
Duplicate #21390
I'm not closing it yet, since the other issue is closed and it's supposed to be fixed in `6.6-beta1`.

**Comment by torkelo** [comment]
will re-open the old issue
